### ISSUE TITLE
Enhanced Table column management

### ISSUE THREAD

**Issue by SeamusLeonardHPE** [grommet/hpe-design-system#3740]
Ladders up to key objective 2 for Q2

Connect with dev to spec how the enhancements to Table column management component in  grommet will impact figma component.
If there is any updates needed visually, compositionally or in documentation we will bring both into alignment.

https://www.figma.com/file/5uOrsL2qk8XwwH8C1ZpDb6/HPE-Design-System-Components?type=design&node-id=2915-56106&mode=design&t=Dvnm5ZvYzKJztVKv-4

**Comment by julauxen** [comment]
After talking to Matt and Taylor, this is the scope for this ticket:

- [ ] Review figma component to allow the checkbox from select columns to be disabled.
- [ ] Review guidelines to include the mandatory column behaviour.

**Comment by julauxen** [comment]
Guidelines update merged, waiting on review on the Figma File

**Comment by SeamusLeonardHPE** [comment]
Passed my review:
Addition of 2 component variants.
Light / Checkbox Checked / Disabled
Dark / Checkbox Checked / Disabled

Julia confirmed with devs and no requirement for 
Light / Checkbox **Unhecked** / Disabled
Dark / Checkbox **Unhecked** / Disabled


![Image](https://github.com/grommet/hpe-design-system/assets/135040188/66407c23-529c-47b1-9a16-91723b2771c2)

Awaiting Olivers reviews and should be ready to publish tomorrow.

**Comment by SeamusLeonardHPE** [comment]
Merged into main and awaiting publish with batch changes.
Tracked in  [#3767](https://github.com/grommet/hpe-design-system/issues/3767)
